cmd/kube-proxy/app/server.go

```diff
@@ -361,6 +364,7 @@ type ProxyServer struct {
 	Client                 clientset.Interface
```

Comment:
Yes, actually I have considered the same thing before. But, it will introduce more changes on iptables & userspace proxy.